### ISSUE TITLE
Institution-specific student ID

### ISSUE THREAD

**Issue by jonatanschroeder** [PrairieLearn/PrairieLearn#7694]
Currently students are identified by UID (based on login) and UIN (internal information used to identify students by the auth provider). However, in some cases (personally both at UBC and York), students have a different identification number, or student ID, that doesn't match either. It would be good to have this information available in some form, either uploaded by the instructor (preferred) or provided by the student. This ID could then be used:

* in the gradebook as a searchable/sortable column, to find a student based on their ID.
* in the students tab for an assessment, for the same reason.
* in the downloads for assessments and/or the gradebook, to simplify handling of these lists in other tools (e.g., import into LMS or spreadsheet-match with imports from other tools).

The field doesn't need to have any formal functionality other than information. I don't have a preference between having this information be linked to a enrolment (course-specific) record or the user (course-agnostic) record, though I imagine the latter may be necessary if instructors are to upload it to avoid confusion.

**Comment by jonatanschroeder** [comment]
UIN is often used as the internal identification used by the auth provider (e.g., PUID in UBC's Shib, subject identifier for Google auth, etc.). At UBC this was not the student number. At York I'm still waiting for them to check my request for Shub, not sure what they use for that. If it's the student ID, I'm happy.

**Comment by nwalters512** [comment]
Yeah... UBC is special, although maybe that will be true of _any_ Canadian university with their unique privacy laws? It's never been perfectly clear to me why UBC Shib has to deal in opaque identifiers.

If we were to do this, I'd be inclined to store this at the enrollment level. Storing it on the user gives a malicious (or just careless) instructor the power to break other courses.

**Comment by firasm** [comment]
To answer your question @nwalters512, I think the reason UBC has opaque identifiers is because Student Numbers are known to students and are likely not kept confidential (apparently). PUID and the like are identifiers associated with their CWL (which used to be PII but now isn't) behind the scenes, but aren't student facing (unless they go digging and look very hard).

So basically, protecting students from themselves 😅. What would make me happy as an instructor (and would solve the issue for me at UBC that Jonatan raised) is just adding the CWL field to Canvas now that it's no longer PII.

<img width="1352" alt="Screenshot 2023-05-23 at 8 52 05 PM" src="https://github.com/PrairieLearn/PrairieLearn/assets/2507459/c4943e23-179d-4d6f-b0b0-efb74bca0778">

- [Source (a remarkably readable document from University Counsel)](https://universitycounsel.ubc.ca/files/2020/01/Fact-Sheet-What-is-Personal-Information.pdf)
